packages/zowe-explorer-api/src/vscode/ui/TableView.ts

```diff
@@ -0,0 +1,571 @@
+                    this.onTableDisplayChangedEmitter.fire(message.data);
+                    return;
+                // "ready" command: The table view has attached its message listener and is ready to receive data.
+                case "ready":
+                    await this.updateWebview();
+                    return;
+                // "copy" command: Copy the data for the row that was right-clicked.
+                case "copy":
+                    await env.clipboard.writeText(JSON.stringify(message.data.row));
+                    return;
+                case "copy-cell":
+                    await env.clipboard.writeText(message.data.cell);
+                    return;
+                default:
+                    break;
+            }
+
+            const row: number = message.rowIndex ?? 0;
+            const matchingActionable = [
+                ...(this.data.actions[row] ?? []),
+                ...this.data.actions.all,
+                ...(this.data.contextOpts[row] ?? []),
+                ...this.data.contextOpts.all,
+            ].find((action) => action.command === message.command);
+            if (matchingActionable != null) {
+                switch (matchingActionable.callback.typ) {
+                    case "single-row":
+                        await matchingActionable.callback.fn(this, { index: message.data.rowIndex, row: message.data.row });
+                        break;
+                    case "multi-row":
+                        await matchingActionable.callback.fn(this, message.data.rows);
+                        break;
+                    case "cell":
+                        await matchingActionable.callback.fn(this, message.data.cell);
+                        break;
+                    // TODO: Support column callbacks? (if there's enough interest)
+                    default:
+                        break;
+                }
+            }
+        }
+
+        /**
+         * (Sender) message handler for the table view.
+         * Used to send data and table layout changes to the table view to be re-rendered.
+         *
+         * @returns Whether the webview received the update that was sent
+         */
+        private async updateWebview(): Promise<boolean> {
+            const result = await (this.panel ?? this.view).webview.postMessage({
+                command: "ondatachanged",
+                data: this.data,
+            });
+
+            if (result) {
+                this.onTableDataReceivedEmitter.fire(this.lastUpdated ? diff(this.lastUpdated, this.data) : this.data);
+                this.lastUpdated = this.data;
+            }
+            return result;
+        }
+
+        /**
+         * Access the unique ID for the table view instance.
+         *
+         * @returns The unique ID for this table view
+         */
+        public getId(): string {
+            if (this.uuid == null) {
+                this.uuid = randomUUID();
+            }
```

Comment:
Not requesting to change anything, but the nullish coalescing assignment operator could be used here 😀 
```suggestion
            this.uuid ??= randomUUID();
```